[docs] Add known limitations to api-routes docs (#24727)
expo/expo: docs/pages/preview/api-routes.mdx

If you're using any environment variables or **.env** files, add them to Netlify. You can do this by going to the **Site settings** and adding them to the **Build & deploy** section.

</Step>

> ## Known Limitations

Reviewer: ```suggestion
## Known limitations
```

There are a number of known features that are not currently supported in the API Routes beta release.

### No dynamic imports

> API Routes currently work by bundling all code (minus the Node.js built-ins) into a single file. This means that you cannot use any external dependencies that are not bundled with the server. For example, a library like `sharp` which includes multiple platform binaries cannot be used. This will be addressed in a future version.

Reviewer: ```suggestion
API Routes currently work by bundling all code (minus the Node.js built-ins) into a single file. This means that you cannot use any external dependencies that are not bundled with the server. For example, a library such as `sharp`, which includes multiple platform binaries, cannot be used. This will be addressed in a future version.
```

Reviewer: Some suggestions on wording as per our writing style guide for concistency.

### ESM not supported

> The current bundling implementation opts to be more unified than flexible. This means the limitation of native not supporting ESM is carried over to API Routes. All code will be transpiled down to Common JS (`require`/`module.exports`), however we recommend you write API Routes using ESM regardless. This will be addressed in a future version.

Reviewer: ```suggestion
The current bundling implementation opts to be more unified than flexible. This means the limitation of native not supporting ESM is carried over to API Routes. All code will be transpiled down to Common JS (`require`/`module.exports`). However, we recommend you write API Routes using ESM regardless. This will be addressed in a future version.
```

Let's split the sentence here.
